src/main/java/data/Vendor.java

```diff
@@ -91,17 +114,20 @@ public void addAccount(Account account) {
   }
 
   /** Return a JSON String representing a billing config's content. */
-  public String buildJsonConfig() {
+  public String toJson() {
     ArrayList<Account> accounts = getAccounts();
-    String config = String.format("{\"legacy_vendor_id\":%s," +
-      "\"next_gen_vendor_id\":%d,\"accounts\":[", getLegacyVendorID(), 
+    String config = String.format("{\"legacy_customer_id\":\"%s\"," +
+      "\"next_gen_customer_id\":%d,\"accounts\":[", getLegacyVendorID(),
       getNextGenVendorID());
 
-    for(int i = 0; i < accounts.size(); i++) {
-      config += accounts.get(i).toJson();
+    for (Account account : accounts) {
+      config += account.toJson() + ",";
+    }
+    // Remove trailing comma from list of accounts for valid JSON
```

Comment:
Another way to fix this fence-post problem would be to have a for-loop that runs while (int i = 0: i < accounts.size() - 1; i++) and then just add the final Account to config without adding a period at the end.

Reply:
+1

Reply:
got it, makes a cleaner solution too